services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationDataMapper.kt

```diff
@@ -2,57 +2,64 @@ package fi.hel.haitaton.hanke.application
-            applicationData.endTime
-                ?: throw AlluDataException("applicationData.endTime", AlluDataError.NULL),
-            applicationData.pendingOnClient,
-            applicationData.identificationNumber,
-            applicationData.clientApplicationKind,
-            getWorkDescription(applicationData),
-            applicationData.contractorWithContacts.toAlluData(
-                "applicationData.contractorWithContacts"
-            ),
-            applicationData.postalAddress?.toAlluData(),
-            applicationData.representativeWithContacts?.toAlluData(
-                "applicationData.representativeWithContacts"
-            ),
-            applicationData.invoicingCustomer?.toAlluData("applicationData.invoicingCustomer"),
-            applicationData.customerReference,
-            applicationData.area,
-            applicationData.propertyDeveloperWithContacts?.toAlluData(
-                "applicationData.propertyDeveloperWithContacts"
-            ),
-            applicationData.constructionWork,
-            applicationData.maintenanceWork,
-            applicationData.emergencyWork,
-            applicationData.propertyConnectivity
-        )
+    private const val BASE = "applicationData."
 
-    private fun getWorkDescription(applicationData: CableReportApplicationData): String {
-        val rockExcavation =
-            applicationData.rockExcavation
-                ?: throw AlluDataException("applicationData.rockExcavation", AlluDataError.NULL)
-        return applicationData.workDescription +
-            if (rockExcavation) "\nLouhitaan" else "\nEi louhita"
-    }
+    private const val AREAS = "areas"
+    private const val CONTRACTOR = "contractorWithContacts"
+    private const val CUSTOMER = "customerWithContacts"
+    private const val DEVELOPER = "propertyDeveloperWithContacts"
+    private const val EXCAVATION = "rockExcavation"
+    private const val INVOICING = "invoicingCustomer"
+    private const val REPRESENTATIVE = "representativeWithContacts"
+    private const val TIME_START = "startTime"
+    private const val TIME_END = "endTime"
+
+    fun toAlluData(
+        hankeTunnus: String,
+        applicationData: CableReportApplicationData
+    ): AlluCableReportApplicationData =
+        with(applicationData) {
+            val description = workDescription(cableReport = this)
+            AlluCableReportApplicationData(
+                name = name,
+                customerWithContacts = customerWithContacts.toAlluData(path(CUSTOMER)),
+                geometry = getGeometry(applicationData = this),
+                startTime = startTime.orThrow(path(TIME_START)),
+                endTime = endTime.orThrow(path(TIME_END)),
+                pendingOnClient = pendingOnClient,
+                identificationNumber = hankeTunnus,
+                clientApplicationKind = description, // intentional
```

Comment:
This change should probably show up somewhere in tests as well.

Reply:
toAlluData is now tested in ApplicationDataMapperTest.